### PR TITLE
adding friends, following, pending to items returned

### PR DESCRIPTION
**Pull request by lanre-ade**

### CODE REVIEW

**Comment by lanre-ade** [comment]
:wave: ready for review

**Comment by derekdowling** [comment]
I'd honestly squish this down into just the AuthorSerializer. If we return some extra info that's alright. Then it's way easier to keep consistent everywhere. The other thing is we also need `"requests":[]` which follows the exact same format.

File: api/serializers/author.py [hunk 1]
```diff
@@ -180,17 +180,21 @@ class Meta:
...
+
 class AuthorSerializer(serializers.ModelSerializer):
     displayname = serializers.CharField(source='user.username')
     email       = serializers.EmailField(source='user.email')
     first_name  = serializers.CharField(source='user.first_name')
     last_name   = serializers.CharField(source='user.last_name')
     image       = AuthorImageSerializer(required = False)
+    friends     = CachedAuthorSerializer(many = True)
+    following   = CachedAuthorSerializer(many = True)
+    pending     = CachedAuthorSerializer(many = True)
```
Comment (KarimBaaba) [inline api/serializers/author.py]: missing `requests` ?

**Comment by derekdowling** [comment]
:+1:

**Comment by xkb** [comment]
`'following', 'pending', 'requests'` shouldn't be returned on every request.

File: api/serializers/author.py [hunk 2]
```diff
@@ -186,11 +186,26 @@ class AuthorSerializer(serializers.ModelSerializer):
...
     class Meta:
         model = Author
         fields = ('id', 'displayname', 'email', 'first_name', 'last_name', \
-            'github_username', 'bio', 'host', 'url', 'image')
+            'github_username', 'bio', 'host', 'url', 'image', 'friends', 'following', 'pending', 'requests')
+
+    def to_representation(self, data):
+        representation = super(AuthorSerializer, self).to_representation(data)
+
+        if hasattr(self.context, 'request') and (self.context.request.user.id == data.id):
```
Comment (lanre-ade) [inline api/serializers/author.py]: should i use a try-catch block instead to figure out if context has request?

**Comment by lanre-ade** [comment]
:wave:

File: api/serializers/author.py [hunk 3]
```diff
@@ -186,11 +186,25 @@ class AuthorSerializer(serializers.ModelSerializer):
...
     class Meta:
         model = Author
         fields = ('id', 'displayname', 'email', 'first_name', 'last_name', \
-            'github_username', 'bio', 'host', 'url', 'image')
+            'github_username', 'bio', 'host', 'url', 'image', 'friends', 'following', 'pending', 'requests')
+
+    def to_representation(self, obj):
+        representation = super(AuthorSerializer, self).to_representation(obj)
+
+        if (not hasattr(self.context, 'request')) or (self.context.request.user.id != obj.user.id):
```
Comment (derekdowling) [inline api/serializers/author.py:202]: Beauty!

**Comment by derekdowling** [comment]
LGTM :+1:

**Comment by xkb** [comment]
:+1:

**Pull request merged**